umpost/um2netcdf.py

```diff
@@ -352,34 +345,61 @@ def process(infile, outfile, args):
             c.cell_methods = fix_cell_methods(c.cell_methods)
 
             try:
-                fix_latlon_coord(c, grid_type, dlat, dlon)
+                fix_latlon_coord(c, mv.grid_type, mv.d_lat, mv.d_lon)
             except iris.exceptions.CoordinateNotFoundError:
                 print('\nMissing lat/lon coordinates for variable (possible timeseries?)\n')
                 print(c)
                 raise Exception("Variable can not be processed")
 
-            fix_level_coord(c, z_rho, z_theta)
+            fix_level_coord(c, mv.z_rho, mv.z_theta)
 
             if do_mask:
                 # Pressure level data should be masked
-                if require_heaviside_uv(c.item_code):
-                    if heaviside_uv:
-                        apply_mask(c, heaviside_uv, args.hcrit)
-                    else:
-                        continue
-
-                if require_heaviside_t(c.item_code):
-                    if heaviside_t:
-                        apply_mask(c, heaviside_t, args.hcrit)
-                    else:
-                        continue
+                if require_heaviside_uv(c.item_code) and heaviside_uv:
+                    apply_mask(c, heaviside_uv, args.hcrit)
+
+                if require_heaviside_t(c.item_code) and heaviside_t:
+                    apply_mask(c, heaviside_t, args.hcrit)
 
             if args.verbose:
                 print(c.name(), c.item_code)
 
+            # TODO: split cubewrite ops into funcs & bring those steps into process() workflow
+            #       or a sub process workflow function (like process_mule_vars())
             cubewrite(c, sman, args.compression, args.use64bit, args.verbose)
 
 
+MuleVars = collections.namedtuple("MuleVars", "grid_type, d_lat, d_lon, z_rho, z_theta")
```

Comment:
For info: `mule` functionality had to be extracted to a sub-workflow simplify test setup (otherwise 4 _more_ mock patches were required). What are folks' thoughts on extracting blocks like this to sub-workflows?

Reply:
I really like it. It's cleaner, it removes non-controversial code out of the main workflow so it's easier to read, and typically makes debugging easier when something explodes (i.e., you get a granulated function/function output to look at, rather than a huge monolothic one).

Reply:
(Also, should have added I much prefer having a single object floating around, instead of a collection of assorted variable names to keep track of.)